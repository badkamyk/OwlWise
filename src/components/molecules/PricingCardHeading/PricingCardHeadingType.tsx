export type PricingCardHeadingType = {
	heading: string;
	paragraph: string;
	price: string;
}