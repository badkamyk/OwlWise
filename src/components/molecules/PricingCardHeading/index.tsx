import ContentHeading from "../../atoms/ContentHeading";
import MainContentParagraph from "../../atoms/MainContentParagraph";
import { PricingCardHeadingType } from "./PricingCardHeadingType";

export default function PricingCardHeading({
	heading,
	paragraph,
	price,
}: PricingCardHeadingType) {
	return (
		<>
			<ContentHeading text={heading} size={"xl"} color={"white"} />
			<MainContentParagraph color={"white"} text={paragraph} />
			<div className="my-8 flex items-baseline justify-center">
				<span className="mr-2 text-5xl font-extrabold text-white">{price}</span>
				<span className="text-gray-500 dark:text-white">/month</span>
			</div>
		</>
	);
}
