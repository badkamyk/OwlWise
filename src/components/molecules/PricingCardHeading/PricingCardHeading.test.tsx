import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import PricingCardHeading from "./index";

it("renders PricingCardHeading with props correctly", () => {
	const { container } = render(
		<PricingCardHeading
			heading={"heading"}
			paragraph={"paragraph"}
			price={"333"}
		/>
	);
	expect(screen.getByText("heading")).toBeTruthy();
	expect(screen.getByText("paragraph")).toBeTruthy();
	expect(screen.getByText("333")).toBeTruthy();
	expect(container.getElementsByClassName("text-xl text-white")).toBeTruthy();
	expect(
		container.getElementsByClassName("my-8 flex items-baseline justify-center")
	).toBeTruthy();
	expect(
		container.getElementsByClassName("mr-2 text-5xl font-extrabold text-white")
	).toBeTruthy();
	expect(
		container.getElementsByClassName("text-gray-500 dark:text-white")
	).toBeTruthy();
	expect(container).toMatchSnapshot();
});
