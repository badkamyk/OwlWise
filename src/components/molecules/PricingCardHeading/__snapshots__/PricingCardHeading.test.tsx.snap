// Vitest Snapshot v1

exports[`renders PricingCardHeading with props correctly 1`] = `
<div>
  <div
    data-testid="pricing-card-heading"
  >
    <h3
      class="mb-4 tracking-tight font-extrabold text-white text-xl"
    >
      heading
    </h3>
    <p
      class="mb-4 text-white text-md"
      data-testid="main-content-paragraph"
    >
      paragraph
    </p>
    <div
      class="my-8 flex items-baseline justify-center"
    >
      <span
        class="mr-2 text-5xl font-extrabold text-white"
      >
        333
      </span>
      <span
        class="text-gray-500 dark:text-white"
      >
        /month
      </span>
    </div>
  </div>
</div>
`;
